src/components/QuranReader/ReflectionView/ReflectionItem.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @next/next/no-img-element */
+import { useState } from 'react';
+
+import useTranslation from 'next-translate/useTranslation';
+
+import OverflowMenuIcon from '../../../../public/icons/menu_more_horiz.svg';
+import VerifiedIcon from '../../../../public/icons/verified.svg';
+
+import styles from './ReflectionItem.module.scss';
+
+import Button, { ButtonShape, ButtonSize, ButtonVariant } from 'src/components/dls/Button/Button';
+import PopoverMenu from 'src/components/dls/PopoverMenu/PopoverMenu';
+import { formatDateRelatively } from 'src/utils/datetime';
+import { getQuranReflectPostUrl } from 'src/utils/navigation';
+import { truncateString } from 'src/utils/string';
+import { navigateToExternalUrl } from 'src/utils/url';
+
+type ReflectionItemProps = {
+  id: number;
+  authorName: string;
+  avatarUrl: string;
+  date: string;
+  reflectionText: string;
+  isAuthorVerified: boolean;
+};
+
+const DEFAULT_IMAGE = '/images/quran-reflect.png';
+const MAX_REFLECTION_LENGTH = 220;
+const ReflectionItem = ({
+  id,
+  authorName,
+  date,
+  avatarUrl,
+  reflectionText,
+  isAuthorVerified,
+}: ReflectionItemProps) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const { t, lang } = useTranslation('common');
+  const formattedDate = formatDateRelatively(new Date(date), lang);
+  const onMoreLessClicked = () => setIsExpanded(!isExpanded);
```

Comment:
setting a state is async so using the current value of `isExpanded` will not guarantee the latest value. we should use it like `setIsExpanded((prevIsExpanded) => !prevIsExpanded)`